flocker/node/agents/blockdevice.py

```diff
@@ -1961,7 +1959,7 @@ def _calculate_desired_state(
         # `ATTACHED_ELSEWHERE` need the same behavior as `NON_MANIFEST`, so we
         # don't check them either.
 
-        not_in_use_datasets = not_in_use(local_datasets.values())
```

Comment:
:question: This one needs a comment and could be bundled into the datastructure-optimizations branch (mentioned above.)